main_server/local_server/server_admin/c_comdef_admin_ajax_handler.class.php

```diff
@@ -334,18 +326,22 @@ private function MarkDeletedMeetingAsHandled($meeting_id)
                 foreach ($obj_array as $change) {
                     if ($change instanceof c_comdef_change and $change->GetChangeType() === 'comdef_change_type_delete' and !$change->GetAfterObject()) {
                         $meeting = $change->GetBeforeObject();
-                        if ($meeting->GetWorldID() === 'deleted') {
-                            return false;
+                        if (!$isServerAdmin && !in_array(intval($meeting->GetServiceBodyID()), $userServiceBodyIDs)) {
+                            // meeting was under some other service body
+                            return 'not_found';
```

Comment:
I think the changes to this file look really good. The only suggestion I'd make is that now that we are returning these `'not_found'`, `'marked'`, and `'not_marked'` strings from a function, maybe I'd store their values in variables. Really, an enum object would be ideal, but the versions of PHP we are using don't have a proper enum.

You can abuse a class to do the same thing.

```
abstract class SomeEnum {
    const NOT_FOUND = 'not_found';
    const MARKED = 'marked';
    const NOT_MARKED = 'not_marked';
}

// and then you could replace all instances of the string with...
SomeEnum::NOT_FOUND
SomeEnum::MARKED
SomeEnum::NOT_MARKED
```